### PR TITLE
Fixes for calculating collisions in LinkBlocked and Enter

### PR DESCRIPTION
**Pull request by boskoramen**
# About the pull request

Refactors logic and updates documentation for turf/Enter() and LinkBlocked() procs. Fixes a few bugs or weird behaviors with old logic.

Bug with windows not being destroyed properly after being run over with a tank in the video has also been fixed

# Explain why it's good for the game

Multitile collision behavior is pretty wonky, this should make tank act as expected when running into multiple blockers and with things that are bordering an edge like barricades

Also fixes issue with not being able to climb diagonally (and other issues that may have been caused by LinkBlocked returning TRUE when it should not be)
# Testing Photographs and Procedure
Tested with tank, climbing diagonally, and walking around barricades.
<details>
<summary>Screenshots & Videos</summary>

Test video with tank

</details>

# Changelog
:cl: TheDonkified
fix: Can now climb on windows diagonally if there are no actual movement blockers
fix: Multitile vehicles should now properly handle situations where a border object (e.g. indestructible window) was blocking its movement
refactor: Cleaned up logic for detecting if a link is blocked or for checking collisions during movement
refactor: Created new multitile element which is used by all multitile objects, so any new multitile mobs, objs, etc need to be through this element. This element handles movement and collisions for multitile objects.
code: Added unit tests for Enter() and LinkBlocked()
/:cl:

### CODE REVIEW

**Comment by boskoramen** [comment]
didn't realize you can rebase onto master in github

File: .vscode/tasks.json [hunk 1]
```diff
@@ -102,6 +102,15 @@
...
             "label": "tgui: rebuild tgfont"
+        },
+        {
+            "type": "shell",
+            "command": "bin/server",
+            "windows": {
+                "command": ".\\bin\\server.cmd"
+            },
+            "group": "build",
+            "label": "run local server"
```
Comment (harryob) [inline .vscode/tasks.json:113]: this is equivalent to the dreammaker - dreamdaemon target?

**Review by harryob — CHANGES_REQUESTED** [review]
also, ci fail

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Review by harryob — CHANGES_REQUESTED** [review]
stale, merge conflicts, etc. would be happy to try and get this moved through if you'd like to keep working on it, just let me know

**Pull request closed without merging**